src/metrics/test_coverage/index.ts

```diff
@@ -0,0 +1,90 @@
+import { logger } from "../../core";
+import { decode } from "js-base64";
+import { keys, includes, find, propEq, toLower} from "ramda";
+import { DataEvent, MetricsSignature } from "../../interfaces";
+import {
+  WorkflowJobTestCoveragePayload,
+  WorkflowJobTestCoverageOutput,
+} from "./interfaces";
+import moment from "moment";
+import octokit from "../../core/octokit";
+import { isWorkflowJob } from "../../signatureConditions";
+
+export const collectWorkflowsTestCoverageMetrics = async (
+  dataEvent: DataEvent
+): Promise<DataEvent> => {
+  const payload = dataEvent.payload as WorkflowJobTestCoveragePayload;
+
+  const repo = payload.repository.name;
+  const owner = payload.repository.owner.login;
+  const completed_at = moment.utc(payload.workflow_job.completed_at).valueOf();
+  const created_at = moment.utc(payload.workflow_job.created_at).valueOf();
+  const started_at = moment.utc(payload.workflow_job.started_at).valueOf();
+  const duration = completed_at - started_at;
+  const status = payload.workflow_job.status;
+  const workflow_name = payload.workflow_job.workflow_name;
+  const steps = payload.workflow_job.steps.map((step) => ({   
+    ...step,
+    started_at: moment.utc(step.started_at).valueOf(),
+    completed_at: moment.utc(step.completed_at).valueOf(),
+    duration:
+      moment.utc(step.completed_at).valueOf() -
+      moment.utc(step.started_at).valueOf(),
+  }));
+
+  const isWorkflowNameTest = includes('test', toLower(workflow_name));
+  const stepRunTest = find(propEq('name', 'Run test'), steps);
+
+  const test_coverage = {
+    workflow_have_test_data: isWorkflowNameTest ? "yes" : "no",
+    test_run_status: stepRunTest?.status,
+    test_run_started_at: stepRunTest?.started_at,
+    test_run_completed_at: stepRunTest?.completed_at,
+    test_run_duration: stepRunTest?.duration,
+    test_run_failed: stepRunTest?.status === "completed" && stepRunTest?.conclusion !== "success" ? "yes" : "no",
+    test_run_conclusion: stepRunTest?.conclusion
+  }
+
+  console.log("testCoverageData", test_coverage);
+
+  let packages = [];
+  try {
+    const getPackageJson = await octokit.request(
+      "GET /repos/{owner}/{repo}/contents/{path}",
+      {
+        owner: owner,
+        repo: repo,
+        path: "package.json",
+      }
+    );
+    packages = JSON.parse(decode(getPackageJson.data["content"]));
+  } catch (e) {
+    packages = [];
+  }
+  const output: WorkflowJobTestCoverageOutput = {
```

Comment:
The ACs for the ticket are: 

- [ ] Track if a workflow that has "test "in the name has failed or not
- [ ] Track script steps that has "test "in the name has failed or not
- [ ] Percentage of test coverage
- [ ] Duration of test

We can provide more data, but only if they are really important and related to the metrics, otherwise are unnecessary. 

In the test the "test_coverage" data (except the 2 booleans), are all undefined: 

`      test_coverage: {
        workflow_have_test_data: 'no',
        test_run_status: undefined,
        test_run_started_at: undefined,
        test_run_completed_at: undefined,
        test_run_duration: undefined,
        test_run_failed: 'no',
        test_run_conclusion: undefined
      }`


- [ ] Track linting if data is available in workflow
- [ ] Track formatting if data is available in workflow

Reply:
The ACs are divided into two list and the follow metrics are collected 
-  Track if a workflow that has "test "in the name has failed or not
-  Track script steps that has "test "in the name has failed or not
-  Duration of test